### PR TITLE
AB#78-code-editing-experience

### PR DESCRIPTION
**Pull request by flipmess**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -21,6 +21,7 @@
     }
   },
   "rules": {
+    "import/no-extraneous-dependencies": ["error", {"packageDir": path.join(__dirname, './')}]
```
Comment (StuckiSimon) [inline .eslintrc]: Note sure why you would need this? But it's certainly not working.

File: game-frontend/src/components/help/index.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const Help: FC = () => {
       <Dialog isOpen={isOpen} onDismiss={close} aria-label="help">
         <DialogWrapper>
           <Button className="close-button" onClick={close}>
-            <span>×</span>
+            <span>Close</span>
```
Comment (StuckiSimon) [inline game-frontend/src/components/help/index.tsx]: Aren't we usually writing things in German in the app? :v: (See Help Dialog)

File: package.json [hunk 3]
```diff
@@ -40,7 +40,10 @@
     "url": "https://github.com/code-and-conquer/darwin/issues"
   },
   "homepage": "https://github.com/code-and-conquer/darwin#readme",
-  "dependencies": {},
+  "dependencies": {
+    "@monaco-editor/react": "^3.1.2",
+    "monaco-editor": "^0.20.0"
```
Comment (StuckiSimon) [inline package.json]: That's the wrong package.json. This must be in `game-frontend/package.json`. (Small hint: look for the package.json containing other runtime dependencies such as react)

File: game-frontend/src/components/user-script/index.tsx [hunk 4]
```diff
@@ -29,7 +31,8 @@ const UserScript: FC = () => {
   return (
     <Container data-testid="user-script-form">
       <form>
-        <Textarea onChange={onChange} rows={8}></Textarea>
```
Comment (StuckiSimon) [inline game-frontend/src/components/user-script/index.tsx:32]: Make sure to update the userScript - atm it's not synced and therefore the code written is lost.

**Review by StuckiSimon — CHANGES_REQUESTED** [review]
Not sure in what state this PR is - left feedback nevertheless.

File: game-frontend/src/components/help/index.tsx (hunk 2 above)
Comment (flipmess) [reply game-frontend/src/components/help/index.tsx]: oops, must have been looking at the 'save' button when i did that.

File: .eslintrc (hunk 1 above)
Comment (flipmess) [reply .eslintrc]: eslint was showing permanent import error for monaco-editor without this. google said i should try this. what's not working?

File: game-frontend/src/components/user-script/index.tsx (hunk 4 above)
Comment (flipmess) [reply game-frontend/src/components/user-script/index.tsx:32]: will do, thx

File: package.json (hunk 3 above)
Comment (flipmess) [reply package.json]: is that the reason eslint was complaining?

File: .eslintrc (hunk 1 above)
Comment (StuckiSimon) [reply .eslintrc]: well, you forgot the trailing comma - so now eslint is simply not working at all.
Which kind of explains why you don't get an error now 😄

File: package.json (hunk 3 above)
Comment (StuckiSimon) [reply package.json]: Yes, usually when you get an error there is a reason - that's the reason 😄 
Add the dependencies you want to `game-frontend/package.json` and you're good to go.

Btw: As far as I'm aware of you don't have to add `monaco-editor` at all :v:

**Comment by michaelschufi** [comment]
Got it finally working with all the react hooks and the typing. 
@flipmess will update the keybindings, so eslint won't complain.

**Comment by sir-marc** [comment]
Closed by #69

**Pull request closed without merging**